graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphqlErrorsHandlers.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.graphql;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+
+import graphql.ExecutionResult;
+import graphql.ExecutionResultImpl;
+import graphql.GraphQLError;
+import graphql.GraphqlErrorException;
+
+final class GraphqlErrorsHandlers {
+
+    private static final Logger logger = LoggerFactory.getLogger(GraphqlErrorsHandlers.class);
+
+    /**
+     * Returns a {@link HttpResponse} based on {@link Throwable} or List of {@link GraphQLError}.
+     * @param errorsMappingFunction The function which maps the {@link GraphQLError} to an {@link HttpStatus}
+     */
+    static GraphqlErrorsHandler of(GraphqlErrorsMappingFunction errorsMappingFunction) {
+        final GraphqlErrorsMappingFunction mappingFunction = withDefault(errorsMappingFunction);
+        return (ctx, input, produceType, executionResult, cause) -> {
+            if (cause != null) {
+                // graphQL.executeAsync() returns an error in the executionResult with getErrors().
+                // Use 500 Internal Server Error because this cause might be unexpected.
+                final ExecutionResult error = newExecutionResult(cause);
+                return HttpResponse.ofJson(HttpStatus.INTERNAL_SERVER_ERROR, produceType,
+                                           error.toSpecification());
+            }
+            final List<GraphQLError> errors = executionResult.getErrors();
+            final HttpStatus httpStatus = mappingFunction.apply(ctx, input, errors);
+            assert httpStatus != null; // withDefault
```

Comment:
There's a chance that the `mappingFunction` returns `null` so we should handle it. 
How about adding `GraphqlErrorsMappingFunction.orElse(default)` method and apply it here?